File: src/weaponinventorymod/gui/WimGuiButtonSpecs.java
package weaponinventorymod.gui;

import java.util.ArrayList;
import java.util.List;

final class WimGuiButtonSpecs {
    private WimGuiButtonSpecs() {
    }

    @SafeVarargs
    static <A> List<WimGuiButtonSpec<A>> of(WimGuiButtonSpec<A>... specs) {
        List<WimGuiButtonSpec<A>> result = new ArrayList<WimGuiButtonSpec<A>>();
        if (specs == null) {
            return result;
        }
        for (int i = 0; i < specs.length; i++) {
            if (specs[i] != null) {
                result.add(specs[i]);
            }
        }
        return result;
    }
}

File: src/main/kotlin/weaponsprocurement/gui/WimGuiButtonSpecs.kt
package weaponsprocurement.gui

import java.util.ArrayList

class WimGuiButtonSpecs private constructor() {
    companion object {
        @JvmStatic
        fun <A> of(vararg specs: WimGuiButtonSpec<A>?): List<WimGuiButtonSpec<A>> {
            val result = ArrayList<WimGuiButtonSpec<A>>()
            for (spec in specs) {
                if (spec != null) result.add(spec)
            }
            return result
        }
    }
}
